vme/src/structs.h

```diff
@@ -70,49 +70,402 @@ class pc_data : public char_data
+    void setSkillLevelAtIndexTo(size_t index, ubit8 value) { m_skill_lvl[index] = value; }
+    /// @}
+
+    /**
+     * @name Weapon skill
+     * @todo Add range checking to indexes
+     * @{
+     */
+    sbit16 getWeaponSkillAtIndex(size_t index) const { return m_weapons[index]; }
+    sbit16 *getWeaponSkillArrayPtr() { return &m_weapons[0]; }
+    sbit16 *getWeaponSkillAtIndexPtr(size_t index) { return &m_weapons[index]; }
+    void setWeaponSkillAtIndexTo(size_t index, sbit16 value) { m_weapons[index] = value; }
+    void increaseWeaponSkillAtIndexBy(size_t index, sbit16 value) { m_weapons[index] += value; }
+    void decreaseWeaponSkillAtIndexBy(size_t index, sbit16 value) { m_weapons[index] -= value; }
+    /// @}
+
+    /**
+     * @name Weapon Skill Level
+     * @todo Add range checking to indexes
+     * @{
+     */
+    ubit8 getWeaponSkillLevelAtIndex(size_t index) const { return m_weapon_lvl[index]; }
+    ubit8 *getWeaponSkillLevelAtIndexPtr(size_t index) { return &m_weapon_lvl[index]; }
+    ubit8 *getWeaponSkillLevelArrayPtr() { return &m_weapon_lvl[0]; }
+    void setWeaponSkillLevelAtIndexTo(size_t index, ubit8 value) { m_weapon_lvl[index] = value; }
+    /// @}
+
+    /**
+     * @name Ability Level
+     * @todo Add range checking to indexes
+     * @{
+     */
+    ubit8 getAbilityLevelAtIndex(size_t index) const { return m_ability_lvl[index]; }
+    ubit8 *getAbilityLevelAtIndexPtr(size_t index) { return &m_ability_lvl[index]; }
+    ubit8 *getAbilityLevelArrayPtr() { return &m_ability_lvl[0]; }
+    void setAbilityLevelAtIndexTo(size_t index, ubit8 value) { m_ability_lvl[index] = value; }
+    /// @}
+
+    /**
+     * @name Conditions
+     * @todo Add range checking to indexes
+     * @{
+     */
+    sbit8 getConditionAtIndex(size_t index) const { return m_conditions[index]; }
+    sbit8 *getConditionAtIndexPtr(size_t index) { return &m_conditions[index]; }
+    void setConditionAtIndexTo(size_t index, sbit8 value) { m_conditions[index] = value; }
+    void increaseConditionAtIndexBy(size_t index, sbit8 value) { m_conditions[index] += value; }
+    void decreaseConditionAtIndexBy(size_t index, sbit8 value) { m_conditions[index] -= value; }
+    /// @}
+
+    /**
+     * @name Access Level
+     * @{
+     */
+    ubit8 getAccessLevel() const { return m_nAccessLevel; }
+    void setAccessLevel(ubit8 value) { m_nAccessLevel = value; }
+    /// @}
+
+    /**
+     * @name Passwords
+     * @{
+     */
+    const char *getPassword() const { return m_pwd; }
+    void setPassword(const char *value)
+    {
+        strncpy(m_pwd, value, PC_MAX_PASSWORD);
+        m_pwd[PC_MAX_PASSWORD - 1] = 0;
+    }
+    int readPasswordFrom(CByteBuffer &buf)
+    {
+        auto retval = buf.ReadStringCopy(m_pwd, PC_MAX_PASSWORD);
+        m_pwd[PC_MAX_PASSWORD - 1] = '\0';
+        return retval;
+    }
+    void truncatePassword()
+    {
+        m_pwd[10] = '\0'; // This will allow me to later extend the password length
```

Comment:
That 10 eventually needs addressing. Is it PC_MAX_PASSWORD.... ?

Reply:
Yeah. I can't remember. There was definitely something tricky about the PWD. I believe that I increased it from 10 -> 13. So either I forgot the [10] or there was some obscure reason that I needed it to be 10.